File: 3Bimestre/Backend/TFI/src/main/frontend/src/services/dentistService.ts
import { HTTPMethods, IDentist } from '../types';
import { fetchAPI } from './fetchAPI';

export const listDentists = async (): Promise<IDentist[]> => {
  return await fetchAPI(HTTPMethods.Get, 'dentists');
};

export const getDentist = async (id: string): Promise<IDentist> => {
  return await fetchAPI(HTTPMethods.Get, 'dentists/' + id);
};

export const createDentist = async (dentist: IDentist): Promise<IDentist> => {
  return await fetchAPI(HTTPMethods.Post, 'dentists/', dentist);
};

export const updateDentist = async (id: string, updatedDentist: IDentist): Promise<IDentist> => {
  return await fetchAPI(HTTPMethods.Put, 'dentists/' + id, updatedDentist);
};

export const deleteDentist = async (id: string): Promise<string> => {
  return await fetchAPI(HTTPMethods.Delete, 'dentists/' + id);
};

File: 3Bimestre/Backend/CTD-Backend/src/TFI/Services/DentistService.java
package TFI.Services;

import TFI.DAO.DAO;
import TFI.Models.Dentist;
import java.util.List;

public class DentistService {
    private final DAO<Dentist> DAO;

    public DentistService(DAO<Dentist> DAO) {
        this.DAO = DAO;
    }

    public Dentist createDentist(Dentist dentist) {
        int id = DAO.create(dentist);
        dentist.setId(id);
        return dentist;
    }

    public Dentist updateDentist(int id, Dentist dentist) {
        return DAO.update(id, dentist);
    }

    public void deleteDentist(int id) {
        DAO.delete(id);
    }

    public Dentist getDentist(int id) {
        return DAO.get(id);
    }

    public List<Dentist> listDentists() {
        return DAO.list();
    }
}
